RHIDP-3874 Authenticating with the guest user (#479)
redhat-developer/red-hat-developers-documentation-rhdh: modules/auth/proc-authenticationg-with-the-guest-user.adoc

> [id="authenticating-with-the-guest-user"]

Reviewer: Is :_mod-docs-content-type: PROCEDURE not used for RHDH?

Author: ```suggestion
:_mod-docs-content-type: PROCEDURE
[id="authenticating-with-the-guest-user_{context}"]
```

> [id="authenticating-with-the-guest-user"]

Reviewer: [id="authenticating-with-the-guest-user_{context}"]

See https://redhat-documentation.github.io/modular-docs/#module_anchor-and-file-names-concept > Anchors

= Authenticating with the Guest user

> To start exploring {product} features, do not worry about authentication: after the default installation, you can authenticate with the Guest user, and use all the application features.

Reviewer: So a user doesn't need to authenticate to explore {product}, but the Prereq require installation. So is this procedure to help them explore the product, or to perform a default installation, and be able to log in as a Guest?

Author: ```suggestion
To start exploring {product} features, do not worry about setting up authentication: after the default installation, you can authenticate with the Guest user, and use all the application features.
```

Reviewer: As a user, why do I want to follow the rest of this procedure if I'm just exploring? I don't quite get the purpose of "To start exploring {product} features, do not worry about setting up authentication" as an intro. It seems unnecessary.

Or do you mean a user doesn't have to setup their company network authentication with RHDH, but instead they can they can create a Guest user, and use Guest to poke around to see if RHDH provides a solution for their need?

.Prerequisites
* You installed {product-short} by using default values.

> See link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{product-version}#Install%20and%20Upgrade[Installing RHDH].

Reviewer: Does the user need to select the installation method they want to use? Or is there a general RHDH installation method, and not one specific to OCP, EKS, air-gapped, etc?

Author: Default install. Regardless of the platform. No need to introduce a non variable here.

.Procedure
. Go to the {product-short} login page.

> . Log in with the Guest user account.

Reviewer: Where do they find the Guest user account information? Or is there a button that says "Guest login" on the login page? It is unclear how they login with the "Guest user" account from this step.

Author: ```suggestion
. To log in with the Guest user account, click **Enter** in the **Guest** tile.
```

.Verification

> * You can use all RHDH features.

Reviewer: Should "RHDH" be {product-short} ?

Author: ```suggestion
* You can use all {product-short} features.
```

:_mod-docs-content-type: PROCEDURE
[id="authenticating-with-the-guest-user_{context}"]
= Authenticating with the Guest user

> To start exploring {product} features, do not worry about setting up authentication: after the default installation, you can authenticate with the Guest user, and use all the application features.

Reviewer 2: Similar to PR #460, I'd suggest something like this...
```suggestion
To explore {product-short} features, you do not need to configure authentication and authorization. By default, {product-short} allows you to authenticate with the Guest user which enables you to access authorized features associated with the guest user role.
```

Author: ```suggestion
To explore {product-short} features, you can skip configuring authentication and authorization. 
After the default installation, you can log in as a Guest user and access all the features.
```

:_mod-docs-content-type: PROCEDURE
[id="authenticating-with-the-guest-user_{context}"]
= Authenticating with the Guest user

To explore {product-short} features, you can skip configuring authentication and authorization. 
After the default installation, you can log in as a Guest user and access all the features.

.Prerequisites
* You installed {product-short} by using default values.
See link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{product-version}#Install%20and%20Upgrade[Installing RHDH].

.Procedure
. Go to the {product-short} login page.
. To log in with the Guest user account, click **Enter** in the **Guest** tile.

.Verification

> * You can use all {product-short} features.

Author: ```suggestion
* In the **Settings** page, your profile name is **Guest**.
* You can use all {product-short} features.
```

:_mod-docs-content-type: PROCEDURE
[id="authenticating-with-the-guest-user_{context}"]
= Authenticating with the Guest user

To explore {product-short} features, you can skip configuring authentication and authorization. 
After the default installation, you can log in as a Guest user and access all the features.

.Prerequisites

> * You installed {product-short} by using default values.

Author: With 1.3 the default installation does not work OOTB. @PatAKnight and @gustavolira to confirm:

```suggestion
* You link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{product-version}/html/administration_guide_for_red_hat_developer_hub/assembly-add-custom-app-file-openshift_admin-rhdh[added a custom {product-short} application configuration], and have sufficient permissions to modify it.
```

See link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{product-version}#Install%20and%20Upgrade[Installing RHDH].

.Procedure
. Go to the {product-short} login page.
. To log in with the Guest user account, click **Enter** in the **Guest** tile.

.Verification

> * In the **Settings** page, your profile name is **Guest**.

Author: ```suggestion
* In the {product-short} **Settings** page, your profile name is **Guest**.
```

> .Procedure
> . Go to the {product-short} login page.
> . To log in with the Guest user account, click **Enter** in the **Guest** tile.
> 
> .Verification
> * In the {product-short} **Settings** page, your profile name is **Guest**.
> * You can use {product-short} features.

Author: ```suggestion
.Procedure

* To set up the guest user in your {product-short} custom configuration, edit your custom Developer Hub config map, such as `app-config-rhdh`, and add following lines to the `app-config-rhdh.yaml` content:
+
.`app-config-rhdh.yaml` fragment
[code,yaml]
----
auth:
  environment: development
  providers:
    guest:
      dangerouslyAllowOutsideDevelopment: true
----

.Verification
. Go to the {product-short} login page.
. To log in with the Guest user account, click **Enter** in the **Guest** tile.
. In the {product-short} **Settings** page, your profile name is **Guest**.
. You can use {product-short} features.
```

> * You can use all {product-short} features.

Author: ```suggestion
* You can use {product-short} features.
```
